terraform/cloudwatch.tf.json.template.py

```diff
@@ -1,34 +1,110 @@
+import json
+
 from azul import (
     config,
+    require,
+)
+from azul.queues import (
+    Queues,
 )
 from azul.terraform import (
     emit_tf,
 )
 
-emit_tf(None if config.disable_monitoring else {
-    "resource": [
+
+def dashboard_body() -> str:
+    # To minify the template and confirm it is valid JSON before deployment we
+    # parse the template file as JSON and then convert it back to a string.
+    with open(config.cloudwatch_dashboard_template) as f:
+        body = json.load(f)
+    body = json.dumps(body)
+
+    def prod_qualified_resource_name(name: str) -> str:
+        resource, _, suffix = config.unqualified_resource_name_and_suffix(name)
+        return config.qualified_resource_name(resource, suffix=suffix, stage='prod')
+
+    queues = Queues()
+    qualified_resource_names = [
+        *config.all_queue_names,
+        *queues.functions_by_queue().values()
```

Comment:
I think we should also assert that the set of keys returned by `functions_by_queue` is a subset of `config.all_queue_names`.